experiment/evaluator.py

```diff
@@ -195,13 +205,78 @@ def check_target(self, ai_binary, target_path: str) -> Optional[Result]:
       traceback.print_exc()
       return None
 
-  def _parse_libfuzzer_logs(self, log_handle) -> tuple[int, int, bool]:
-    """Parse libFuzzer logs."""
+  def _parse_stacks_from_libfuzzer_logs(self, lines: list[str]) -> list[str]:
+    """Parse stack traces from libFuzzer logs."""
+    # There can have over one thread stack in a log.
+    stacks = []
+
+    # A stack -> a sequence of stack frame lines.
+    stack, stack_parsing = [], False
+    for line in lines:
+      is_stack_frame_line = LIBFUZZER_STACK_FRAME_LINE_PREFIX.match(
+          line) is not None
+      if (not stack_parsing) and is_stack_frame_line:
+        # First line.
+        stack_parsing = True
+        stack = [line.strip()]
+      elif stack_parsing and is_stack_frame_line:
+        # Middle line(s).
+        stack.append(line.strip())
+      elif stack_parsing and (not is_stack_frame_line):
+        # Last line.
+        stack_parsing = False
+        stacks.append(stack)
+
+    # Last stack.
+    if stack_parsing:
+      stacks.append(stack)
+
+    return stacks
+
+  def _parse_fuzz_cov_info_from_libfuzzer_logs(
+      self,
+      lines: list[str]) -> tuple[Optional[int], Optional[int], Optional[int]]:
+    """Parses cov of INITED & DONE, and round number from libFuzzer logs."""
+    initcov, donecov, lastround = None, None, None
+
+    for line in lines:
+      if line.startswith('#'):
+        # Parses cov line to get the round number.
+        match = LIBFUZZER_COV_LINE_PREFIX.match(line)
+        roundno = int(match.group(1)) if match else None
+
+        if roundno is not None:
+          lastround = roundno
+          if 'INITED' in line:
+            initcov = int(line.split('cov: ')[1].split(' ft:')[0])
+          elif 'DONE' in line:
+            donecov = int(line.split('cov: ')[1].split(' ft:')[0])
+
+    return initcov, donecov, lastround
+
+  def _stack_func_is_of_testing_project(self, stack_frame: str) -> bool:
+    return bool(CRASH_STACK_WITH_SOURCE_INFO.match(stack_frame)) and (
```

Comment:
This incurred the following error for the same reason as [the incorrect type above](https://github.com/google/oss-fuzz-gen/pull/187/files#r1546206543).
```python
    return bool(CRASH_STACK_WITH_SOURCE_INFO.match(stack_frame)) and (
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: expected string or bytes-like object, got 'list'
```

Could you please fix this in a new PR (and branch)? Thanks!